tests/ui/components/navigation/navigationcomponent.js

```diff
@@ -102,3 +112,131 @@ describe('navigation component configuration', () => {
     expect(tab).toMatchObject(expected);
   });
 });
+
+describe('navigation tab links are correct', () => {
+  let COMPONENT_MANAGER;
+
+  const defaultConfig = {
+    container: '#test-component',
+    verticalKey: 'verticalKey',
+    verticalPages: [
+      { verticalKey: 'vertical', url: './vertical', label: 'Vertical' },
+      { verticalKey: 'facets', url: './facets', label: 'Facets' }
+    ]
+  };
+
+  beforeEach(() => {
+    COMPONENT_MANAGER = mockManager({
+      globalStorage: new GlobalStorage(),
+      persistentStorage: new PersistentStorage()
+    });
+
+    COMPONENT_MANAGER.getComponentNamesForComponentTypes = () => [];
+  });
+
+  it('tab links contain the referrerPageUrl from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.REFERRER_PAGE_URL, 'yext.com');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('referrerPageUrl=yext.com');
+  });
+
+  it('tab links contain the context from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.API_CONTEXT, 'some context');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('context=some+context');
+  });
+
+  it('updating the context in global storage updates the context in tab links', () => {
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.API_CONTEXT, 'new context');
+
+    // Re-render because the component state changed
+    wrapper.update();
```

Comment:
shouldn't the component rerender on it's own when API_CONTEXT is updated?

Reply:
*I looked a little more into this (trying to write some unit tests of my own) and I think in the past we never tried to test how the component reacts to state changes in global storage (since we were using a mocked out one) and would set the state prior to mounting. A few tests do simulate clicks, and I'm not quite sure how those are working, but I think it could be related to AnswersAdapter (enzymeadapter.js)

Reply:
Based on the jsdoc for `update()`, it seems like this is an appropriate use case for it. Here's what the jsdoc says: 'Forces a re-render. Useful to run before checking the render output if something external may be updating the state of the component somewhere.'

I also did some manual testing and it doesn't cause setState() to be called additional times which suggests that the answers component re-render lifecycle isn't occurring. So I believe it's just the ReactWrapper component updating rather than the answers component updating

Reply:
Yeah, I think this is functionality our testing framework is lacking is lacking